src/pages/staticPages/tools/oncoprinter/Oncoprinter.tsx

```diff
@@ -379,9 +399,168 @@ export default class Oncoprinter extends React.Component<
         return WindowStore.size.width - 75;
     }
 
+    @action.bound
+    public handleSelectedClinicalTrackColorChange(
+        value: string,
+        color: RGBAColor | undefined
+    ) {
+        if (this.selectedClinicalTrack) {
+            this.props.store.setUserSelectedClinicalTrackColor(
+                this.selectedClinicalTrack.label,
+                value,
+                color
+            );
+        }
+    }
+
+    @observable trackKeySelectedForEdit: string | null = null;
+
+    @action.bound
+    setTrackKeySelectedForEdit(key: string | null) {
+        this.trackKeySelectedForEdit = key;
+    }
+
+    @computed get selectedClinicalTrack() {
+        return _.find(
+            this.props.store.clinicalTracks,
+            t => t.key === this.trackKeySelectedForEdit
+        );
+    }
+
+    @computed get defaultClinicalAttributeColoringForStringDatatype() {
+        if (this.selectedClinicalTrack) {
+            return _.mapValues(
+                getDefaultClinicalAttributeColoringForStringDatatype(
+                    this.selectedClinicalTrack.data
+                ),
+                hexToRGBA
+            );
+        }
+        return _.mapValues(RESERVED_CLINICAL_VALUE_COLORS, hexToRGBA);
+    }
+
+    @autobind
+    private getDefaultSelectedClinicalTrackColor(value: string) {
+        if (!this.selectedClinicalTrack) {
+            return DEFAULT_WHITE;
+        }
+        switch (this.selectedClinicalTrack.datatype) {
+            case 'counts':
+                return MUTATION_SPECTRUM_FILLS[
+                    _.indexOf(MUTATION_SPECTRUM_CATEGORIES, value)
+                ];
+            case 'string':
+                // Mixed refers to when an event has multiple values (i.e. Sample Type for a patient event may have both Primary and Recurrence values)
+                if (value === 'Mixed') {
+                    return [220, 57, 18, 1];
+                } else {
+                    return this
+                        .defaultClinicalAttributeColoringForStringDatatype[
+                        value
+                    ];
+                }
+            default:
+                return DEFAULT_WHITE;
+        }
+    }
+
+    @computed get selectedClinicalTrackValues() {
+        if (this.selectedClinicalTrack) {
+            return getClinicalTrackValues(this.selectedClinicalTrack);
+        }
+        return [];
+    }
+
     public render() {
         return (
             <div className="posRelative">
+                {this.selectedClinicalTrack && (
```

Comment:
seems not good that we are duplicating this code (presumably it also exists in results view codebase?